vms/platformvm/block/executor/backend.go

```diff
@@ -95,3 +99,28 @@ func (b *backend) getTimestamp(blkID ids.ID) time.Time {
 	// so we just return the chain time.
 	return b.state.GetTimestamp()
 }
+
+// VerifyUniqueInputs returns nil iff no blocks in the inclusive
+// ancestry of [blkID] consume an input in [inputs].
+func (b *backend) VerifyUniqueInputs(blkID ids.ID, inputs set.Set[ids.ID]) error {
```

Comment:
q: why is this exported? It wasn't. Are we planning to use it outside P-chain block executor package?

Reply:
It doesn't need to be. I felt exporting it makes it clearer that it can be consumed by structs outside of `backend`, namely in the `verifier`